test/debug_info/redundant_inst.f90

```diff
@@ -0,0 +1,20 @@
+!RUN: %flang %s -g -S -emit-llvm -o - | FileCheck %s --check-prefix=STORE
+!RUN: %flang %s -g -o %t
+!RUN: llvm-dwarfdump --debug-line %t | FileCheck %s --check-prefix=LINETABLE
+
+!Check that `store` instruction is getting emitted for the second assignment.
+!STORE: store i32 4, i32* %[[VAR_A:.*]], align 4
+!STORE: %[[TEMP:.*]] = load i32, i32* %[[VAR_A]], align 4
+!STORE: store i32 %[[TEMP]], i32* %[[VAR_A]], align 4
+
+!Check the line table entry of the second assignment.
+!LINETABLE: Address    Line Column File ISA Discriminator Flags
+!LINETABLE: 0x{{.*}}    18    1      1   0         0       is_stmt
```

Comment:
The dwarfdump test fails for me in our downstream compiler. The following is the table generated. The STORE check in LLVM IR passes.

```
Address            Line   Column File   ISA Discriminator Flags
------------------ ------ ------ ------ --- ------------- -------------
0x0000000000400964     15      0      1   0             0  is_stmt
0x0000000000400980     17      1      1   0             0  is_stmt prologue_end
0x0000000000400984     19      1      1   0             0  is_stmt
0x00000000004009c8      0      1      1   0             0
0x00000000004009cc     19      1      1   0             0
0x00000000004009dc     20      1      1   0             0  is_stmt
0x00000000004009e8     20      1      1   0             0  is_stmt end_sequence
```

Reply:
Thanks for the feedback @kiranchandramohan :)
Based on your shared `llvm-dwarfdump` log, it's not possible to backtrack which entry is getting contributed from the `store` that this patch adds. 
Could you please share the `!dbg !N` and associated metadata attached to the `store` instruction ?
For this case, since CI passes the test, I'll assume it should me something like:
`!13 = !DILocation(**line: 18**, column: 1, scope: !10)` // lineno is what we are primarily interested in.
If you're noticing lineno other then `18` with your downstream compiler, then AFAICT compiler somehow filled incorrect lineno -- one possible reason could be some delta between upstream and downstream :(
I mentioned this delta thing because the test is passing correctly with upstream CI.

Reply:
The store has the correct debug info. It is being lost somewhere in isel. We have the fast-isel at O0 I think. I was able to kind of reproduce the failure on AArch64 with the following commands. I did not try on X86_64.

```
/home/kiran/open_source_flang/install/bin/flang /home/kiran/open_source_flang/flang/test/debug_info/redundant_inst.f90 -g -S -emit-llvm -o - | /home/kiran/open_source_flang/install/bin/llc  -filetype=obj -o - | llvm-dwarfdump --debug-line -
```

If you can do either of the following then I can approve.
1) Remove the dwarfdump portion and just check the metadata and line number in the emitted LLVM IR.
2) Use fglobal-isel for the second flang invocation to generate the object file.

Reply:
I think this is a good time to enable Aarch* builds in CI ?
> If you can do either of the following then I can approve.

Remove the dwarfdump portion and just check the metadata and line number in the emitted LLVM IR.
Use fglobal-isel for the second flang invocation to generate the object file.

Okay here's my observations:
`-fast-isel=true` -- passes on x86-64 -- Used in this patch
`-fast-isel=false` -- fails on on x86-64 (I think it's taking SelectionDAG Isel)
`-fast-isel=false -global-isel=true` -- Crashing (Only community `flang` our downstream is happy with this option too).

Since `Aarch` is more inclined towards `global-isel`, I think `-fast-isel=true` or `-global-isel=false` should solve the problem.
In this revision, I've use `-fast-isel=true`, please let me know how this works out :)